resolwe/storage/connectors/googleconnector.py

```diff
@@ -0,0 +1,83 @@
+"""Google storage connector"""
+import base64
+from google.cloud import storage
+
+from .baseconnector import BaseStorageConnector
+
+
+class GoogleConnector(BaseStorageConnector):
+    """Google Cloud Storage storage connector"""
+
+    def __init__(self, config: dict, name: str):
+        super().__init__(config, name)
+        self.bucket_name = config["bucket"]
+        self.supported_upload_hash = ["crc32c", "md5"]
+        self.supported_download_hash = ["crc32c", "md5"]
+        self.hash_propery = {"md5": "md5_hash", "crc32c": "crc32c"}
+
+    def get_object_list(self, url):
+        if url != "" and not url.endswith("/"):
+            url += "/"
+        return [e.name for e in self.bucket.list_blobs(prefix=url)]
+
+    def _initialize(self):
+        """Construct and return client to interface with Google Storage from
+        given credentials.
+
+        :param credentials: filename of the stored credentials.
+        :type credentials: str
+        """
+        credentials = self.config["credentials"]
+        self._client = storage.Client.from_service_account_json(credentials)
+        self._bucket = self._client.get_bucket(self.bucket_name)
+
+    def __getattr__(self, name):
+        requires_initialization = ["client", "bucket"]
+        if name in requires_initialization:
```

Comment:
Since connectors are registered when storage application is loaded the __init__ method is always called when some management command is executed.

It makes some calls to the google/S3 and I wanted to avoid that.